libssh-rs/src/channel.rs

```diff
@@ -46,6 +46,9 @@ impl Drop for Channel {
             // Prevent any callbacks firing as part the remainder of this drop operation
             sys::ssh_remove_channel_callbacks(self.chan_inner, self._callbacks.as_mut());
         }
+        if self.chan_inner.is_null() {
```

Comment:
this should be ordered ahead of the `sys::ssh_remove_channel_callbacks` call above.

However, I think this may not be needed if `forget` is used in the alternative destruction path.

Reply:
Removed in favor of `mem::forget`.